shared/src/commonMain/kotlin/com/karumi/gallery/app/PhotoListPresenter.kt

```diff
@@ -1,48 +1,42 @@
 package com.karumi.gallery.app
 
-import com.karumi.gallery.logError
-import com.karumi.gallery.logInfo
-import com.karumi.gallery.model.PhotoShot
+import com.karumi.gallery.model.Photos
 import com.karumi.gallery.usecase.GetPhotos
+import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.collect
+import kotlin.coroutines.CoroutineContext
 
 class PhotoListPresenter(
   private val view: View,
   private val getAllPhotos: GetPhotos
-) {
+) : CoroutineScope {
 
-  companion object {
-    private const val TAG = "PhotoListPresenter"
-  }
+  override val coroutineContext: CoroutineContext
+    get() = job
 
-  private var getPhotosJob: Job? = null
-
-  fun onCreate() {
-    view.showLoader()
-    getPhotosJob = launchInMain {
-      try {
-        logInfo(TAG, "Start getting photos")
```

Comment:
Do we lose the logs? I'd prefer to keep those logs to understand in common module how is going the code in both platforms (android or iOS)

Reply:
Nevermind 👍